src/intelli/node/objectsink.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * GTlab IntelliGraph
+ *
+ *  SPDX-License-Identifier: BSD-3-Clause
+ *  SPDX-FileCopyrightText: 2024 German Aerospace Center
+ *
+ *  Author: Jens Schmeink <jens.schmeink@dlr.de>
+ */
+#include "objectsink.h"
+
+#include <intelli/data/object.h>
+
+#include <QPushButton>
+
+#include "gt_application.h"
+#include "gt_datamodel.h"
+#include "gt_project.h"
+
+using namespace intelli;
+
+ObjectSink::ObjectSink() :
+    Node("Object sink"),
+    m_target("target", tr("Target"), tr("Target"), QString(), this,
+               QStringList() << GT_CLASSNAME(GtObject), true)
+{
+    m_in  = addInPort({typeId<ObjectData>(), tr("Object")});
+
+    registerProperty(m_target);
+
```

Comment:
Since the node is not doing much in its `eval` method, you may  use ´Blocking` Flag do avoid having to create a copy of the node just to execute the few lines of code. 

My recommendation is to remove the `eval` method entirely and subscribe to `inputDataRecieved`. Please also use `Blocking` in this case).

```suggestion

setNodeEvalMode(NodeEvalMode::Blocking);

```